src/theory/arith/theory_arith_private.cpp

```diff
@@ -3938,15 +3938,36 @@ Node TheoryArithPrivate::branchIntegerVariable(ArithVar x) const {
   TNode var = d_partialModel.asNode(x);
   Integer floor_d = d.floor();
 
-  //Node eq = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::EQUAL, var, mkRationalNode(floor_d+1)));
-  //Node diseq = eq.notNode();
+  if (options::cubeTest())
+  {
+    Trace("integers") << "cube test enabled"  << endl;
+    Integer ceil_d = d.ceiling();
+    Rational f = r - floor_d;
+    // Multiply by -1 to get abs value.
+    Rational c = (r - ceil_d) * (-1); 
+ 
+    Integer nearest;
+    (c > f) ? nearest = floor_d : nearest = ceil_d;
 
-  Node ub = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::LEQ, var, mkRationalNode(floor_d)));
-  Node lb = ub.notNode();
+    Node ub = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::LEQ, var, mkRationalNode(nearest - 1)));
+    Node lb = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::GEQ, var, mkRationalNode(nearest + 1)));
+    Node lem = NodeManager::currentNM()->mkNode(kind::OR, ub, lb);
+    Node eq = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::EQUAL, var, mkRationalNode(nearest)));
+    Node literal = d_containing.getValuation().ensureLiteral(eq);
+    d_containing.getOutputChannel().requirePhase(literal, true);
+    Node lem = NodeManager::currentNM()->mkNode(kind::OR, literal, lem);
+  }
+  else
+  {
+    //Node eq = Rewriter::rewrite(NodeManager::currentNM()->mkNode(kind::EQUAL, var, mkRationalNode(floor_d+1)));
```

Comment:
You can remove this commented out code.

Reply:
This commented out code was there when I checked out this file, which is why I left it. I'm happy to remove it, but not sure if someone wanted it to remain there.

Reply:
I think its fine to remove (its been our policy to not have any dead/commented out code in master).